CMakeLists.txt

```diff
@@ -141,7 +141,12 @@ if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
         $<$<COMPILE_LANGUAGE:CXX>:-Wextra>
         $<$<COMPILE_LANGUAGE:CXX>:-pipe>)
 elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
-    message(FATAL_ERROR "Clang is currently not supported.")
+    target_compile_options(${PROJECT_NAME} PUBLIC
+        $<$<COMPILE_LANGUAGE:CXX>:-std=c++20>
+        $<$<COMPILE_LANGUAGE:CXX>:-fexceptions>
+        $<$<COMPILE_LANGUAGE:CXX>:-Wall>
+        $<$<COMPILE_LANGUAGE:CXX>:-Wextra>
```

Comment:
Can you please add a GitHub action which installs clang 16, probably on latest Ubuntu stable 22.04 and builds and runs the tests? I'm definitely not comfortable adding clang as a supported compiler without the tests running on each PR moving forward.

Reply:
I've added github action which installs clang 16 and 17 on ubuntu-latest to build and run the tests